llm_demo/orchestrator/langgraph/langgraph_orchestrator.py

```diff
@@ -46,131 +49,125 @@
 }
 
 
-class UserAgent:
+class UserState(TypedDict):
+    messages: Annotated[Sequence[BaseMessage], add_messages]
     client: ClientSession
-    agent: AgentExecutor
-
-    def __init__(
-        self,
-        client: ClientSession,
-        agent: AgentExecutor,
-        memory: ConversationBufferMemory,
-    ):
-        self.client = client
-        self.agent = agent
-        self.memory = memory
-
-    @classmethod
-    def initialize_agent(
-        cls,
-        client: ClientSession,
-        tools: List[StructuredTool],
-        history: List[BaseMessage],
-        prompt: ChatPromptTemplate,
-        model: str,
-    ) -> "UserAgent":
-        llm = VertexAI(max_output_tokens=512, model_name=model, temperature=0.0)
-        memory = ConversationBufferMemory(
-            chat_memory=ChatMessageHistory(messages=history),
-            memory_key="chat_history",
-            input_key="input",
-            output_key="output",
-        )
-        agent = initialize_agent(
-            tools,
-            llm,
-            agent=AgentType.STRUCTURED_CHAT_ZERO_SHOT_REACT_DESCRIPTION,
-            memory=memory,
-            handle_parsing_errors=True,
-            max_iterations=3,
-            early_stopping_method="generate",
-            return_intermediate_steps=True,
-        )
-        agent.agent.llm_chain.prompt = prompt  # type: ignore
-        return UserAgent(client, agent, memory)
-
-    async def close(self):
-        await self.client.close()
-
-    async def invoke(self, prompt: str) -> Dict[str, Any]:
-        try:
-            response = await self.agent.ainvoke({"input": prompt})
-        except Exception as err:
-            raise HTTPException(status_code=500, detail=f"Error invoking agent: {err}")
-        return response
-
-    async def insert_ticket(self, params: str):
-        return await insert_ticket(self.client, params)
-
-    def reset_memory(self, base_message: List[BaseMessage]):
-        self.memory.clear()
-        self.memory.chat_memory = ChatMessageHistory(messages=base_message)
+    is_last_step: IsLastStep
 
 
 class LangGraphOrchestrator(BaseOrchestrator):
-    _user_sessions: Dict[str, UserAgent]
+    _user_sessions: List[str]
     # aiohttp context
     connector = None
 
     def __init__(self):
-        self._user_sessions = {}
+        self._user_sessions = []
 
     @classproperty
     def kind(cls):
         return "langgraph"
 
+    async def create_graph(self, tools, prompt: ChatPromptTemplate, model_name: str):
```

Comment:
should this be an internal function? it's not on the orchestrator interface. Are we calling it publicly?

Reply:
Moved and updated.